xrpl/asyncio/wallet/wallet_generation.py

```diff
@@ -57,7 +59,9 @@ async def generate_faucet_wallet(
 
     .. # noqa: DAR402 exception raised in private method
     """
-    faucet_url = get_faucet_url(client.url, faucet_host)
+    if not client.network_id:
+        await _get_network_id_and_build_version(client)
```

Comment:
```suggestion
    await _get_network_id_and_build_version(client)
```
It's already in the helper function.